### DISCUSSION TITLE
FAQ

### DISCUSSION THREAD

**Discussion by doitandbedone** [doitandbedone/ispyagentdvr-docker#146] [Q&A]
Please make sure to consult the guides:

Readme:
https://github.com/doitandbedone/ispyagentdvr-docker/blob/master/README.md

Installation:
https://www.ispyconnect.com/download.aspx

User Guide:
https://www.ispyconnect.com/userguide-agent-dvr.aspx

Changelogs:
https://www.ispyconnect.com/producthistory.aspx?productid=27

**Got a question you can't find an answer to? Please create your question here:**
https://github.com/doitandbedone/ispyagentdvr-docker/discussions/new

Note: this conversation is locked, please open a new discussion for better tracking.

**Comment by giube** [comment]
Hello, thanks a lot for the docker!
I'm running it on the Container Station onto a QNAP NAS.
How to setup a local path as a storage path of the recordings of the cameras? Differently to the normale windows version (I installed also on a PC to compare), there isn't the option "Network storage" in the Server settings --> Backup.
I tried with something like /Agent/Dock_Rec/   or /192.168.x.x/Agent/Dock_Rec  (putting the real IP address of the NAS) but I cant' see the recordings there.
I see the recordings in the "Recordings" list, but I don't know where the files are located.
Thanks!

**Reply by doitandbedone** [reply]
Docker works with volumes. You'll see the volume mapping within the run command -v host:container such as `-v /appdata/ispyagentdvr/media/:/agent/Media/WebServerRoot/Media/`
where it maps the docker container's folder /agent/Media/WebServerRoot/Media and all its content to your host machine at /appdata/ispyagentdvr/media.

All videos are stored in /agent/Media/WebServerRoot/Media/video where you'll see the subfolders for each camera, so in your host machine you'll see it in /appdata/ispyagentdvr/media/video or wherever mapping you put.

If you want to add new directory or subdirectory you'll have to recreate the container and add a mapping, in you case add 
`-v /host/machine/directory/:/docker/container/directory/`

**Reply by HSB-collab** [reply]
Hi thanks for the Docker Container. Like @giube above, I would like to be able to write my camera recordings to a samba network share. I have set up the samba share using the docker command with the -v option to map my network share to the docker directory location for the media as follows: 
`docker -v //LAN_network_address_of_share/media:/agent/Media/WebServerRoot/Media/`
When I list the volumes in the docker container (using `docker inspect -f '{{ .Mounts }}' containerid`), I get the following:
`{bind  /LAN_network_address_of_share/media /agent/Media/WebServerRoot/Media   true rprivate}`
Note that there is only one forward slash in front of the network address in the bind. I am not sure if this is significant? I also don't know the meaning of 'rprivate'.
However, when I record a video in Agent DVR, it only records to the directory in the Docker container. I have checked that the network share is mounted and that it is writable from the host machine on which the Agent DVR is running (I can write files using File Manager on the docker host), and the permissions are correct.
I have spent hours trying to figure this out but with no success so any suggestions would be greatly appreciated!

**Reply by doitandbedone** [reply]
@HSB-collab sorry for the delayed response,  I have created a new discussion, just easier to keep track of. Please follow it here:
https://github.com/doitandbedone/ispyagentdvr-docker/discussions/175

**Please note that this conversation is locked and you won't be able to reply, please use the new discussion mentioned above.**

**Comment by doitandbedone** [comment]
## Docker Volumes

Possible scenarios:

- I added a folder/subfolder within the Media volume but I can't see its contents outside the container:
Docker volumes are a bit tricky but nothing impossible to work with. Volumes usually map the top or root level. If you want to add folders or subfolders it must be specified during the run/start command just add the following to the run command:
`-v host/directory/path:container/directory/path`

- I have a volume I want to use on other containers but they can't access it.

More on volumes:
Docker official docs:
https://docs.docker.com/storage/

Sharing volumes between containers:
https://www.digitalocean.com/community/tutorials/how-to-share-data-between-docker-containers

Volume types:
https://spin.atomicobject.com/2019/07/11/docker-volumes-explained/

More on bind mounts:
https://docs.docker.com/storage/bind-mounts/
